Informative error messages for Try-assertions

In case of failure the assertion
```java
Try<Void> result = someFunction();
VavrAssertions.assertThat(result).isSuccess();
```
outputs
```
java.lang.AssertionError: 
Expecting Try to be a Success, but wasn't
```
Most of the time I need to know exactly what went wrong. It would be great if the error message contains a summary of the thrown exception. I'm thinking of its name, message, and stack trace.

What do you think about that?

--- a/src/main/java/org/assertj/vavr/api/AbstractTryAssert.java
+++ b/src/main/java/org/assertj/vavr/api/AbstractTryAssert.java
@@ -246,7 +246,10 @@ abstract class AbstractTryAssert<SELF extends AbstractTryAssert<SELF, VALUE>, VA
 
     private void assertIsSuccess() {
         isNotNull();
-        if (actual.isEmpty()) throwAssertionError(shouldBeSuccess());
+        if (actual.isEmpty()) {
+            final Throwable cause = actual.getCause();
+            throwAssertionError(shouldBeSuccess(cause.getClass(), cause.getMessage(), cause.getStackTrace()));
+        }
     }
 
     private void assertIsFailure() {

--- a/src/main/java/org/assertj/vavr/api/TryShouldBeSuccess.java
+++ b/src/main/java/org/assertj/vavr/api/TryShouldBeSuccess.java
@@ -14,6 +14,8 @@ package org.assertj.vavr.api;
 
 import org.assertj.core.error.BasicErrorMessageFactory;
 
+import java.io.StringWriter;
+
 import static java.lang.String.format;
 
 /**
@@ -23,8 +25,18 @@ import static java.lang.String.format;
  */
 class TryShouldBeSuccess extends BasicErrorMessageFactory {
 
-    private TryShouldBeSuccess() {
-        super(format("%nExpecting Try to be a Success, but wasn't"));
+    private TryShouldBeSuccess(Class<? extends Throwable> clazz, String message, StackTraceElement[] stackTrace) {
+        super(format("%nExpecting Try to be a Success, but was a Failure:" +
+                "%n- exception class: %s" +
+                "%n- message: %s" +
+                "%n- stack trace:\n%s", clazz.getName(), message, asString(stackTrace)));
+    }
+
+    private static String asString(StackTraceElement[] stackTrace) {
+        StringWriter s = new StringWriter();
+        for (StackTraceElement traceElement : stackTrace)
+            s.append("\tat ").append(String.valueOf(traceElement)).append("\n");
+        return s.toString();
     }
 
     /**
@@ -33,7 +45,7 @@ class TryShouldBeSuccess extends BasicErrorMessageFactory {
      * @return a error message factory.
      * @throws java.lang.NullPointerException if Try is null.
      */
-    static TryShouldBeSuccess shouldBeSuccess() {
-        return new TryShouldBeSuccess();
+    static TryShouldBeSuccess shouldBeSuccess(Class<? extends Throwable> clazz, String message, StackTraceElement[] stackTrace) {
+        return new TryShouldBeSuccess(clazz, message, stackTrace);
     }
 }

--- a/src/test/java/org/assertj/vavr/api/TryAssert_containsInstanceOf_Test.java
+++ b/src/test/java/org/assertj/vavr/api/TryAssert_containsInstanceOf_Test.java
@@ -25,7 +25,12 @@ class TryAssert_containsInstanceOf_Test {
                         .containsInstanceOf(NullPointerException.class)
         )
                 .isInstanceOf(AssertionError.class)
-                .hasMessage("\nExpecting Try to be a Success, but wasn't");
+                .hasMessageMatching(
+                        "\\nExpecting Try to be a Success, but was a Failure:\\n" +
+                        "- exception class: java\\.lang\\.NullPointerException\\n" +
+                        "- message: null\\n" +
+                        "- stack trace:\\n" +
+                        "(\\tat .*(\\n)?)+");
     }
 
     @Test

--- a/src/test/java/org/assertj/vavr/api/TryAssert_hasValueSatisfying_Condition_Test.java
+++ b/src/test/java/org/assertj/vavr/api/TryAssert_hasValueSatisfying_Condition_Test.java
@@ -28,7 +28,12 @@ class TryAssert_hasValueSatisfying_Condition_Test {
                 () -> assertThat(Try.failure(new NullPointerException())).hasValueSatisfying(passingCondition)
         )
                 .isInstanceOf(AssertionError.class)
-                .hasMessage("\nExpecting Try to be a Success, but wasn't");
+                .hasMessageMatching(
+                        "\\nExpecting Try to be a Success, but was a Failure:\\n" +
+                        "- exception class: java\\.lang\\.NullPointerException\\n" +
+                        "- message: null\\n" +
+                        "- stack trace:\\n" +
+                        "(\\tat .*(\\n)?)+");
     }
 
     @Test

--- a/src/test/java/org/assertj/vavr/api/TryAssert_hasValueSatisfying_Test.java
+++ b/src/test/java/org/assertj/vavr/api/TryAssert_hasValueSatisfying_Test.java
@@ -27,7 +27,12 @@ class TryAssert_hasValueSatisfying_Test {
                 })
         )
                 .isInstanceOf(AssertionError.class)
-                .hasMessage("\nExpecting Try to be a Success, but wasn't");
+                .hasMessageMatching(
+                        "\\nExpecting Try to be a Success, but was a Failure:\\n" +
+                        "- exception class: java\\.lang\\.NullPointerException\\n" +
+                        "- message: null\\n" +
+                        "- stack trace:\\n" +
+                        "(\\tat .*(\\n)?)+");
     }
 
     @Test

--- a/src/test/java/org/assertj/vavr/api/TryAssert_isSuccess_Test.java
+++ b/src/test/java/org/assertj/vavr/api/TryAssert_isSuccess_Test.java
@@ -21,10 +21,15 @@ class TryAssert_isSuccess_Test {
     @Test
     void should_fail_when_try_is_failure() {
         assertThatThrownBy(
-                () -> assertThat(Try.failure(new NullPointerException())).isSuccess()
+                () -> assertThat(Try.failure(new NullPointerException("NPE error!"))).isSuccess()
         )
                 .isInstanceOf(AssertionError.class)
-                .hasMessage("\nExpecting Try to be a Success, but wasn't");
+                .hasMessageMatching(
+                        "\\nExpecting Try to be a Success, but was a Failure:\\n" +
+                        "- exception class: java\\.lang\\.NullPointerException\\n" +
+                        "- message: NPE error!\\n" +
+                        "- stack trace:\\n" +
+                        "(\\tat .*(\\n)?)+");
     }
 
     @Test

